src/ResetPasswordHelperInterface.php

```diff
@@ -6,13 +6,38 @@
 
 /**
  * @author Jesse Rushlow <jr@rushlow.dev>
- * @author Ryan Weaver <weaverryan@gmail.com>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
  */
 interface ResetPasswordHelperInterface
 {
+    /**
+     * Generate & persist a new password reset token that can be provided to the user
+     *
+     * ResetPasswordHelper persists a ResetPasswordRequest object which contains
+     * the hashed token. The ResetPasswordToken object returned by this method
+     * contains the user token which is used to select and verify a persisted
+     * request object.
+     */
     public function generateResetToken(object $user): ResetPasswordToken;
 
+    /**
+     * Validate a reset request and fetch the user from persistence
+     *
+     * The token provided to the user from generateResetToken() is validated
+     * against a ResetPasswordRequest object stored in persistence.
```

Comment:
I think (but tell me if I'm underthinking a detail) that the `ResetPasswordRequest` object is entirely an implementation detail of how we are storing things with `Doctrine` and not actually relevant to this interface at all. What I mean is, you could implement this interface fully and correctly and never use or store a `ResetPasswordRequest` object.

Reply:
done